microservices-connector/internal/controller/gmconnector_controller.go

```diff
@@ -423,6 +427,36 @@ func reconcileRouterService(ctx context.Context, client client.Client, graph *mc
 			obj.SetNamespace(graph.Namespace)
 		}
 
+		if routerSvcName != "" && routerSvcName != DefaultRouterServiceName {
+			if obj.GetKind() == Service {
+				service_obj := &corev1.Service{}
+				err = scheme.Scheme.Convert(obj, service_obj, nil)
+				if err != nil {
+					return fmt.Errorf("failed to convert unstructured to service: %v", err)
```

Comment:
it's better we add service/deployment name here, since it's not added reconcileResource neither, I will add it in next PR.